src/quo/components/utilities/social/view.cljs

```diff
@@ -0,0 +1,61 @@
+(ns quo.components.utilities.social.view
+  (:require [quo.components.utilities.social.loader :as social-loader]
+            [react-native.core :as rn]
+            [schema.core :as schema]
+            [utils.number]))
+
+(def ?schema
+  [:=>
+   [:cat
+    [:map {:closed true}
+     [:accessibility-label {:optional true} [:or keyword? string?]]
+     [:size {:optional true :default 32} [:or keyword? pos-int?]]
+     [:social {:optional true} [:or keyword? string?]]
+     [:style {:optional true} map?]
+     ;; Ignores `social` and uses this as parameter to `rn/image`'s source.
+     [:image-source {:optional true} [:or string? map?]]
+     ;; If true, adds `data:image/png;base64,` as prefix to the string passed as `image-source`
+     [:add-b64-prefix? {:optional true} boolean?]]]
+   :any])
+
+(defn- size->number
+  "Remove `size-` prefix in size keywords and returns a number useful for styling."
+  [size]
+  (-> size name (subs 5) utils.number/parse-int))
+
+(defn- social-style
+  [style size]
+  (let [size-number (if (keyword? size)
+                      (size->number size)
+                      size)]
+    (assoc style
+           :width  size-number
+           :height size-number)))
+
+(def ^:private b64-png-image-prefix "data:image/png;base64,")
+
+(defn view-internal
+  "Render a social image.
+   Props:
+   - accessibility-label accessibility-label to the rn/image
+   - style:              extra styles to apply to the `rn/image` component.
+   - size:               `:size-nn` or just `nn`, being `nn` any number. Defaults to 32.
+   - social:             string or keyword, it can contain upper case letters or not.
```

Comment:
I think it would be nice to restrict this more as commented below -
https://github.com/status-im/status-mobile/pull/18080/files#r1416055307